galsim/base.py

```diff
@@ -1462,10 +1462,12 @@ def drawKImage(self, re=None, im=None, nx=None, ny=None, bounds=None, scale=None
 
         return re,im
 
-    # Quick and dirty.  Just check reprs are equal.
-    def __eq__(self, other): return repr(self) == repr(other)
+    def __eq__(self, other):
```

Comment:
>And the only way they could possibly end up different is for the user to have explicitly added or deleted it from one of them somewhere along the way.
 
That was my thought here.  Maybe there's a case for comparing the `noise` in the future though if we ever introduce competing noise correlation function estimators for `RealGalaxy`s.